cql2pgjson/src/test/java/org/folio/cql2pgjson/util/DbSchemaUtilsTest.java

```diff
@@ -70,4 +75,38 @@ public void testGetDbIndexForNullTable() {
     assertFalse(dbIndex.isOther());
   }
 
+  @Test
+  public void testFindForeignKeys() throws Exception {
+    // pathA: f -> e -> d -> c -> b -> a
+    // pathB: f -> e -> c -> b -> a
+    String schemaPath = "templates/db_scripts/foreignKeyPath.json";
+    String dbJson = ResourceUtil.asString(schemaPath, CQL2PgJSON.class);
+    Schema dbSchema = ObjectMapperTool.getMapper().readValue(dbJson, org.folio.rest.persist.ddlgen.Schema.class);
+
+    // the shorter path
+    List<DbFkInfo> list = DbSchemaUtils.findForeignKeysFromSourceTableToTargetAlias(dbSchema, "f", "a");
+    assertEquals("f", list.get(0).getTable());
+    assertEquals("e", list.get(1).getTable());
+    assertEquals("c", list.get(2).getTable());
+    assertEquals("b", list.get(3).getTable());
+
+    // target alias
+    list = DbSchemaUtils.findForeignKeysFromSourceTableToTargetAlias(dbSchema, "f", "bAlias");
+    assertEquals("f", list.get(0).getTable());
+    assertEquals("e", list.get(1).getTable());
+    assertEquals("c", list.get(2).getTable());
+
+    // source alias
+    list = DbSchemaUtils.findForeignKeysFromSourceAliasToTargetTable(dbSchema, "eAlias", "b");
+    assertEquals("e", list.get(0).getTable());
+    assertEquals("d", list.get(1).getTable());
+    assertEquals("c", list.get(2).getTable());
+
+    // source alias 2
+    list = DbSchemaUtils.findForeignKeysFromSourceAliasToTargetTable(dbSchema, "e2Alias", "b");
+    list.forEach(fk -> System.out.println(fk));
```

Comment:
Can you remove this line? assert will report any mismatch.

Reply:
done